lib/PuppeteerSharp.Tests/Network/NetworkEventTests.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Network
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class NetworkEventTests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public async Task PageEventsRequest()
+        {
+            var requests = new List<Request>();
+            Page.RequestCreated += (sender, e) => requests.Add(e.Request);
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            Assert.Single(requests);
+            Assert.Equal(TestConstants.EmptyPage, requests[0].Url);
+            Assert.Equal(ResourceType.Document, requests[0].ResourceType);
+            Assert.Equal("GET", requests[0].Method);
```

Comment:
I wouldn't like to introduce a dependency to MVC to have the Verb Enum (https://msdn.microsoft.com/en-us/library/system.web.mvc.httpverbs(v=vs.118).aspx)

But it would be great to have an enum for this.

Reply:
could we use `System.Net.Http.HttpMethod` since we already use `System.Net.HttpStatusCode`?

Reply:
hmmm, after changing it to `System.Net.Http.HttpMethod` the json serializer throw an exception on it since it's not a string.
I ended up creating a simple JsonConverter for the `Method` property